### DISCUSSION TITLE
Is it possible to transfer without Serializations-Deserializations?

### DISCUSSION THREAD

**Discussion by wjhtinger** [eProsima/Fast-DDS#3663] [General]
How to config or code？

**Comment by JesusPoderoso** [comment]
There is a tutorial in this [section of the documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/zero_copy/zero_copy.html#use-case-zero-copy). Also there is a [complete example in our repository](https://github.com/eProsima/Fast-DDS/tree/master/examples/cpp/dds/ZeroCopyExample).

**Reply by wjhtinger** [reply]
So，when Zero-Copy communication used, the Serializations-Deserializations are bypassed?

**Reply by JesusPoderoso** [reply]
When loan mechanism is used, the Serializations-Deserializations are bypassed.
The use of loans plus Data-sharing or intra-process delivery, results in a zero copies communication.
But you can use loans anyways with UDP or Shared memory to avoid Serializations-Deserializations.

**Reply by Syamkamalnadhc** [reply]
Is ROS2 Humble supports the loaning mechanism with UDP transport?

**Reply by JesusPoderoso** [reply]
Hi @Syamkamalnadhc, as answered in [your previous thread](https://github.com/eProsima/Fast-DDS/discussions/5297#discussioncomment-10961293), ROS 2 Humble does not support loan mechanism if data-sharing is disabled. For that reason, loans and UDP communication are not compatible in ROS 2 Humble.
